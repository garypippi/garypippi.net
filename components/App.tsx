import { PropsWithChildren } from 'react'
import 'tailwindcss/tailwind.css'

export const App = (props: PropsWithChildren<any>) => {
    return (
        <div className="container mx-auto">
            {props.children}
        </div>
    )
}
