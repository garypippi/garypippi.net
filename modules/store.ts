import { createStore } from 'redux'
import { useSelector } from 'react-redux'
import { sns } from './ss'

interface AppState {
    snsLinks: sns.SnsLink[]
}

export const useStore = (initialState?: AppState) => createStore<AppState, any, any, any>(
    state => ({
        snsLinks: initialState?.snsLinks || state?.snsLinks || []
    })
)

export const useSnsLinks = () => useSelector<AppState,sns.SnsLink[]>(
    state => state.snsLinks
)
