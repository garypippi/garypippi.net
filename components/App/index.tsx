import { PropsWithChildren } from 'react'
import Link from 'next/link'
import { css } from 'goober'
import { color, lg, mergeClass } from '../../modules/css'
import { getEnv } from '../../modules/env'

const { github, twitter } = getEnv()

interface Props {
    className?: string
}

export const App = ({ className, children }: PropsWithChildren<Props>) => {
    return (
        <div className={mergeClass(className, css`
            padding: 40px 0;
        `)}>
            {children}
            <div className={css`
                margin: 80px auto 0;
                padding-top: 10px;
                max-width: ${lg}px;
                border-top: 1px solid ${color['grey-3']};
            `}>
                <Link
                    href="/"
                    className={css`
                        font-size: 16px;
                        color: ${color['grey-1']};
                `}>
                    {'garypippi.net'}
                </Link>
                <span className={css`
                    margin-left: 10px;
                `}>
                    {'|'}
                </span>
                <a
                    href={github}
                    className={css`
                        font-size: 16px;
                        margin-left: 10px;
                        color: ${color['grey-1']};
                `}>
                    {'Github'}
                </a>
                <a
                    href={twitter}
                    className={css`
                        font-size: 16px;
                        margin-left: 10px;
                        color: ${color['grey-1']};
                `}>
                    {'Twitter'}
                </a>
            </div>
        </div>
    )
}
