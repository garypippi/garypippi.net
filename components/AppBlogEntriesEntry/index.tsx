import Link from 'next/link'
import { css } from 'goober'
import { color } from '../../modules/css'
import { Entry } from '../../modules/markdown/types'
import { format } from 'date-fns'

interface Props {
    entry: Omit<Entry, 'body'>
    className?: string
}

export const AppBlogEntriesEntry = ({
    className = '',
    entry: { href, attr },
}: Props) => {
    return (
        <div
            className={
                className +
                ' ' +
                css`
                    display: inline-block;
                    padding: 10px;
                `
            }
        >
            <Link
                href={href}
                className={css`
                    text-decoration: none;
                `}
            >
                <h3
                    className={css`
                        font-weight: normal;
                        font-size: 18px;
                        margin-bottom: 4px;
                        color: ${color['grey-5']};
                    `}
                >
                    {attr.title}
                </h3>
            </Link>
            <span
                className={css`
                    font-size: 16px;
                    color: ${color['grey-1']};
                `}
            >
                {format(new Date(attr.date), 'yyyy/MM/dd HH:mm')}
            </span>
            {attr.tags.map((tag, i) => (
                <Link
                    key={i}
                    href={`/tags/${tag}`}
                    className={css`
                        font-size: 16px;
                        margin-left: 4px;
                        color: ${color['grey-1']};
                    `}
                >
                    {`#${tag}`}
                </Link>
            ))}
        </div>
    )
}
