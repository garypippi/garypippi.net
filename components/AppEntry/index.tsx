import Link from 'next/link'
import { Entry } from '../../modules/markdown/types'
import { css } from 'goober'
import { theme } from '../../modules/csstheme'
import { format } from 'date-fns'

interface Props {
    entry: Entry
}

export const AppEntry = ({ entry: { href, attr } }: Props) => {
    return (
        <Link
            href={href}
            className={css`
                display: block;
                text-decoration: none;
                &:not(:last-child) {
                    margin-bottom: 16px;
                }
        `}>
            <h3 className={css`
                font-size: 20px;
                font-weight: normal;
                margin-bottom: 8px;
                color: ${theme.color.text};
            `}>
                {attr.title}
            </h3>
            <span className={css`
                font-size: 16px;
                color: ${theme.color.text};
            `}>
                {format(new Date(attr.date), 'yyyy/MM/dd HH:mm')}
            </span>
            {attr.tags.map((tag, i) => (
                <span
                    key={i}
                    className={css`
                    font-size: 14px;
                    padding: 2px 4px;
                    margin-left: 4px;
                    color: ${theme.color.text};
                    background-color: ${theme.color.tag};
                `}>
                    {tag}
                </span>
            ))}
        </Link>
    )
}
