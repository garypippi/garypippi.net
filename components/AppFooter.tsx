import { AppIcon } from './AppIcon'
import { useSnsLinks } from '../modules/store'

export const AppFooter = () => {
    return (
        <div className="2xl:px-64 xl:px-48 lg:px-16 md:px-8 px-4 my-8 py-8 text-gray-400">
            <div className="flex flex-row justify-center items-center">
                {useSnsLinks().map(({ href, icon }, i) => href && (
                    <a
                        key={`footer-sns-links-${i}`}
                        href={href}
                        className="mr-2"
                    >
                        <AppIcon
                            size={24}
                            className="inline"
                        >
                            {icon}
                        </AppIcon>
                    </a>
                ))}
            </div>
        </div>
    )
}
