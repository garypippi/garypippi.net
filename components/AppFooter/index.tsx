import Link from 'next/link'
import { AppIcon } from '../AppIcon'
import { useSnsLinks } from '../../modules/store'

export const AppFooter = () => {
    return (
        <div className="2xl:px-64 xl:px-48 lg:px-16 md:px-8 px-4 my-8 py-8 text-gray-400">
            <div className="flex flex-row justify-center items-center my-4">
                {useSnsLinks().map(({ href, icon }, i) => href && (
                    <a key={`footer-sns-links-${i}`} href={href} className="mr-2 last:mr-0">
                        <AppIcon size={24}>{icon}</AppIcon>
                    </a>
                ))}
            </div>
            <div className="flex justify-center items-center border-b my-4 pb-4">
                <Link href="/about">
                    <a>
                        About
                    </a>
                </Link>
            </div>
            <div className="flex justify-center items-center my-4 text-xs">
                <Link href="/pp">
                    <a>
                        プライバシーポリシー
                    </a>
                </Link>
            </div>
        </div>
    )
}
