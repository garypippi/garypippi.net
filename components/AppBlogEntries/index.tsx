import { css } from 'goober'
import { lg, md } from '../../modules/css'
import { Entry } from '../../modules/markdown/types'
import { AppBlogEntriesEntry } from '../AppBlogEntriesEntry'

interface Props {
    entries: Omit<Entry,'body'>[]
}

export const AppBlogEntries = ({ entries }: Props) => {
    return (
        <div className={css`
            max-width: ${lg}px;
            margin: 0 auto;
        `}>
            {entries.map((entry, i) => (
                <AppBlogEntriesEntry
                    key={i}
                    entry={entry}
                    className={css`
                        width: calc(50% - 10px);
                        margin-bottom: 10px;
                        &:not(:nth-of-type(2n)) {
                            margin-right: 10px;
                        }
                `}/>
            ))}
        </div>
    )
}
