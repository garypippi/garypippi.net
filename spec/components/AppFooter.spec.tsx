import { render } from 'react-dom'
import { mkcontainer } from '../mkcontainer'
import { AppFooter } from '../../components/AppFooter'
import * as Store from '../../modules/store'

const snsLinks = [
    { href: 'hoge', icon: 'fuga' },
    { href: 'fuga', icon: 'hoge' },
]

describe('AppFooter.tsx', () => {
    it('should render', () => {
        const container = mkcontainer()
        const spy = jest.spyOn(Store, 'useSnsLinks')
        spy.mockImplementation(() => snsLinks)
        render(<AppFooter />, container)
        expect(container.innerHTML).toMatchSnapshot()
    })
})
