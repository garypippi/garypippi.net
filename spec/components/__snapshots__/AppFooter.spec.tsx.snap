// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter.tsx should render 1`] = `"<div class=\\"2xl:px-64 xl:px-48 lg:px-16 md:px-8 px-4 my-8 py-8 text-gray-400\\"><div class=\\"flex flex-row justify-center items-center my-4\\"><a href=\\"hoge\\" class=\\"mr-2 last:mr-0\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" class=\\"inline \\"><path d=\\"fuga\\" class=\\"fill-current\\"></path></svg></a><a href=\\"fuga\\" class=\\"mr-2 last:mr-0\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" class=\\"inline \\"><path d=\\"hoge\\" class=\\"fill-current\\"></path></svg></a></div><div class=\\"flex justify-center items-center border-b my-4 pb-4\\"><a href=\\"/about\\">About</a></div><div class=\\"flex justify-center items-center my-4 text-xs\\"><a href=\\"/pp\\">プライバシーポリシー</a></div></div>"`;
